packages/core/src/interpreter.ts

```diff
@@ -758,6 +758,22 @@ export class Interpreter<
       );
     }
 
+    if (!events.length) {
+      const currentState = this.state;
+      const nextState: AnyState = new State({
+        ...currentState,
+        history: currentState,
+        actions: [],
+        transitions: []
+      });
```

Comment:
note that somewhat weirdly this `nextState` will hold a reference to the `currentState._event` because we don't really have any new value to assign to it.